site-generator/src/main/resources/staticresources/js/search.js

```diff
@@ -0,0 +1,154 @@
+(() => {
+    "use strict";
+
+    window.addEventListener("load", () => {
+        const searchResults = document.querySelector("#search-results");
+        const searchModal = document.querySelector("#search-modal");
+        const searchButton = document.querySelector("#search-button");
+        const searchData = document.querySelector("#search-data");
+        searchModal.addEventListener("hide.bs.modal", clearResults);
+        searchModal.addEventListener("shown.bs.modal", () => {
+            searchData.focus();
+        });
+
+        /**
+         * Find the context path
+         * @param contextPath the context path to check
+         * @returns {Promise<* | Response | String>}
+         */
+        const findContextPath = (contextPath) => {
+            let url;
+            if (contextPath.endsWith("/")) {
+                url = contextPath + "search-index.json";
+            } else {
+                url = contextPath + "/search-index.json";
+            }
+            return fetch(url)
+                .then(response => {
+                    if (response.ok) {
+                        sessionStorage.setItem("contextPath", contextPath);
+                        return contextPath;
+                    } else if (response.status === 404) {
+                        const i = contextPath.lastIndexOf("/");
+                        if (i > 0) {
+                            let newContext = contextPath.substring(0, i);
+                            return findContextPath(newContext);
+                        } else {
+                            return "/";
+                        }
+                    }
+                })
+                .catch(error => {
+                    console.error("Failed find context path. ", error);
+                });
+        };
+
+        // Find the context path
+        let contextPathPromise;
+        if (!sessionStorage.getItem("contextPath")) {
+            let contextPath = window.location.pathname;
+            if (contextPath.endsWith("/index.html")) {
+                contextPath = contextPath.substring(0, contextPath.lastIndexOf("/index.html"));
+            }
+            contextPathPromise = findContextPath(contextPath);
+        } else {
+            contextPathPromise = Promise.resolve(sessionStorage.getItem("contextPath"));
+        }
+
+        let index;
+        if (sessionStorage.getItem("wildscribe-index")) {
```

Comment:
This should definitely be versioned for cases when you might navigate a different version in the same session.